src/main/java/io/gatling/plugin/BatchEnterprisePluginClient.java

```
@@ -145,4 +158,26 @@ private SimulationStartResult createAndStartSimulation(
       throw new SimulationStartException(simulation, e);
     }
   }
+
+  private String simulationClassName(Simulation simulation, File file, String simulationClass)
+      throws EnterprisePluginException {
+    final List<String> discoveredSimulationClasses = simulationFullyQualifiedNamesFromFile(file);
+
+    if (discoveredSimulationClasses.isEmpty()) {
+      throw new NoSimulationClassNameFoundException();
+    } else if (simulationClass != null) {
+      if (!discoveredSimulationClasses.contains(simulationClass)) {
+        throw new InvalidSimulationClassException(discoveredSimulationClasses, simulationClass);
+      }
+      return simulationClass;
+    } else if (simulation != null && discoveredSimulationClasses.contains(simulation.className)) {
+      return simulation.className;
+    } else if (discoveredSimulationClasses.size() == 1) {
+      String head = discoveredSimulationClasses.get(0);
+      logger.info("Pick only available simulation class name " + head);
+      return head;
+    }
```

If I remember well, throwing an exception act also like a break, the remaining part of the method is not executed.

What do you think about doing your checks in a "fail-fast" mode ? I think it can improve readability here.

Something like 
```java
if (discoveredSimulationClasses.isEmpty()) {
  throw new NoSimulationClassNameFoundException();
} 
if (otherCheck) {
  throw new OtherException
}
//end of check, now your logic.
```

It avoids chaining nested if / else.

WDYT ?

---

In one hand, we are used to functional paradigm, hence the functional style that consider the if as an expression.
In the other hand, most present `if`s should be guard (and as you wrote, fail fast)

I'm on two minds about this one.

---

no veto for this code that is good, it's just that it would have been clearer to me.